web-access-app/src/main/java/edu/unc/lib/boxc/web/access/controllers/FullRecordController.java

```diff
@@ -219,7 +221,40 @@ public String getFullRecord(String pidString, Model model, HttpServletRequest re
         }
 
         if (briefObject.getResourceType().equals(searchSettings.resourceTypeAggregate)) {
-            model.addAttribute("viewerNeeded", accessCopiesService.hasViewableFiles(briefObject, principals));
+            boolean imageViewerNeeded = accessCopiesService.hasViewableFiles(briefObject, principals);
+            model.addAttribute("imageViewerNeeded", imageViewerNeeded);
+
+            boolean pdfViewerNeeded = false;
+
+            if (!imageViewerNeeded) {
+                // Check if primary object is a pdf
+                pdfViewerNeeded = accessCopiesService.hasViewablePdf(briefObject, principals);
+
+                // Check if first child is a pdf
+                if (!pdfViewerNeeded) {
+                    // Get child objects
+                    SearchRequest searchRequest = new SearchRequest();
+                    SearchState searchState = searchStateFactory.createSearchState();
+                    searchState.setFacetsToRetrieve(null);
+                    searchState.setIgnoreMaxRows(true);
+                    searchRequest.setSearchState(searchState);
+                    searchRequest.setAccessGroups(principals);
+                    searchRequest.setRootPid(pid);
+                    searchRequest.setApplyCutoffs(true);
+                    SearchResultResponse resultResponse = queryLayer.performSearch(searchRequest);
+
+                    if (resultResponse.getResultCount() == 1) {
+                        ContentObjectRecord childObject = resultResponse.getResultList().get(0);
+                        pdfViewerNeeded = accessCopiesService.hasViewablePdf(childObject, principals);
+
+                        if (pdfViewerNeeded) {
+                            model.addAttribute("filePid",childObject.getId());
```

Comment:
I'd prefer if we scoped the name of this attribute, `filePid` could collide with other attributes in the future. Renaming it something like `pdfViewerPid` would work.